gnovm/pkg/gnolang/preprocess.go

```diff
@@ -296,10 +303,12 @@ func initStaticBlocks(store Store, ctx BlockNode, bn BlockNode) {
 					}
 					n.Predefine(false, pte.Name)
 				}
-				for _, rte := range n.Type.Results {
-					if rte.Name != "" {
-						n.Predefine(false, rte.Name)
+				for i, rte := range n.Type.Results {
+					if rte.Name == "" {
+						rn := fmt.Sprintf(".res_%d", i)
+						rte.Name = Name(rn)
```

Comment:
define in `initStaticBlocks` phase to preserve original order, which will impact defer 
behavior.